File: repos_estudiantes_TARQUI_tercera_entrega/ROJAS_ALCON_VICTOR_ALEJANDRO/generecidad/ej5/ej5.py
from typing import Generic, TypeVar, List, Optional

T = TypeVar('T')

class Pila(Generic[T]):
    def __init__(self):
        self.elementos: List[T] = []

    def apilar(self, item: T):
        self.elementos.append(item)

    def desapilar(self) -> Optional[T]:
        if self.elementos:
            return self.elementos.pop()
        return None

    def mostrar(self):
        print("Contenido de la pila (cima al final):")
        for item in reversed(self.elementos):
            print(item)

pila_texto = Pila[str]()
pila_texto.apilar("uno")
pila_texto.apilar("dos")
pila_texto.mostrar()
print("Desapilado:", pila_texto.desapilar())
pila_texto.mostrar()

pila_numeros = Pila[int]()
pila_numeros.apilar(10)
pila_numeros.apilar(20)
pila_numeros.mostrar()
print("Desapilado:", pila_numeros.desapilar())
pila_numeros.mostrar()

File: repos_estudiantes_TARQUI_tercera_entrega/ROJAS_ALCON_VICTOR_ALEJANDRO/generecidad/ej5/ej5.java
import java.util.ArrayList;

class Pila<T> {
    private ArrayList<T> elementos = new ArrayList<>();
    public void apilar(T item) {
        elementos.add(item);
    }

    public T desapilar() {
        if (!elementos.isEmpty()) {
            return elementos.remove(elementos.size() - 1);
        }
        return null; 
    }

    public void mostrar() {
        System.out.println("Contenido de la pila:");
        for (int i = elementos.size() - 1; i >= 0; i--) {
            System.out.println(elementos.get(i));
        }
    }
}

public class Main {
    public static void main(String[] args) {
        Pila<String> pilaTexto = new Pila<>();
        pilaTexto.apilar("uno");
        pilaTexto.apilar("dos");
        pilaTexto.mostrar();
        System.out.println("Desapilado: " + pilaTexto.desapilar());
        pilaTexto.mostrar();

        Pila<Integer> pilaNumeros = new Pila<>();
        pilaNumeros.apilar(10);
        pilaNumeros.apilar(20);
        pilaNumeros.mostrar();
        System.out.println("Desapilado: " + pilaNumeros.desapilar());
        pilaNumeros.mostrar();
    }
}